prov/efa/src/rxr/rxr_pkt_cmd.c

```diff
@@ -449,6 +450,70 @@ ssize_t rxr_pkt_wait_handshake(struct rxr_ep *ep, fi_addr_t addr, struct rxr_pee
 	return 0;
 }
 
+static inline
+void rxr_pkt_handle_data_copied(struct rxr_ep *ep,
+				struct rxr_pkt_entry *pkt_entry)
+{
+	int pkt_type;
+
+	pkt_type = rxr_get_base_hdr(pkt_entry->pkt)->type;
+
+	switch(pkt_type) {
+	case RXR_DATA_PKT:
+	case RXR_READRSP_PKT:
+		rxr_data_pkt_handle_data_copied(ep, pkt_entry);
+		break;
+	case RXR_EAGER_MSGRTM_PKT:
+	case RXR_EAGER_TAGRTM_PKT:
+		rxr_pkt_handle_eager_rtm_data_copied(ep, pkt_entry);
+		break;
+	case RXR_MEDIUM_MSGRTM_PKT:
+	case RXR_MEDIUM_TAGRTM_PKT:
+		rxr_pkt_handle_medium_rtm_data_copied(ep, pkt_entry);
+		break;
+	case RXR_LONG_MSGRTM_PKT:
+	case RXR_LONG_TAGRTM_PKT:
+		rxr_pkt_handle_long_rtm_data_copied(ep, pkt_entry);
+		break;
+	case RXR_EAGER_RTW_PKT:
+		rxr_pkt_handle_eager_rtw_data_copied(ep, pkt_entry);
+		break;
+	case RXR_LONG_RTW_PKT:
+		rxr_pkt_handle_long_rtw_data_copied(ep, pkt_entry);
+		break;
+	default:
+		FI_WARN(&rxr_prov, FI_LOG_CQ,
+			"invalid control pkt type %d\n",
+			rxr_get_base_hdr(pkt_entry->pkt)->type);
+		assert(0 && "invalid control pkt type");
+		rxr_cq_handle_cq_error(ep, -FI_EIO);
+	}
+}
+
+void rxr_pkt_proc_data(struct rxr_ep *ep,
```

Comment:
Function documentation?  Also, there's no verb in the function name; I'm not sure exactly what this function does from the name (other than take an obscene number of arguments).

Reply:
renamed this function to ```rxr_pkt_copy_to_rx()``` in new revision.